### PR TITLE
BAU: Upgrade nginx and changes to nginx config file

### PR DESCRIPTION
**Pull request by adityapahuja**
This pull request contains three commits.

1. The first commit upgrades nginx to the latest version v1.17.5 and it brings in bug fixes, security fix and new features.
2. The second commit removes the need to create soft links to /dev/stdout and /dev/stderr manually. The official nginx image already creates symbolic links to  /dev/stdout and /dev/stderr. It also sets worker_processes to auto.
3. The third commit updates nginx to use strong TLS and ciphers.

Please see commits for further information.

Author: @adityapahuja

### CODE REVIEW

File: dockerfiles/nginx-tls/nginx.conf.tpl [hunk 1]
```diff
@@ -20,8 +19,8 @@ http {
 
   server {
     listen       8443 ssl;
-    ssl_protocols       TLSv1.1 TLSv1.2;
-    ssl_ciphers         ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128:AES256:HIGH:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4:!kEDH;
+    ssl_protocols       TLSv1.2 TLSv1.3;
```
Comment (jakubmiarka) [inline dockerfiles/nginx-tls/nginx.conf.tpl]: is this going to cut-off old browsers not supporting 1.2?

Comment (adityapahuja) [reply dockerfiles/nginx-tls/nginx.conf.tpl]: No, according to Kibana, none of hub components are using TLSv1.2. Nginx is internal and is not facing public Internet directly.

Comment (adityapahuja) [reply dockerfiles/nginx-tls/nginx.conf.tpl]: I mean TLS v1.1.

Comment (jakubmiarka) [reply dockerfiles/nginx-tls/nginx.conf.tpl]: ah ok, I wasn't sure whether the user traffic was going through nginx or not

**Pull request merged**